tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -202,62 +259,140 @@ public Void call() throws Exception {
                         latencyHistogram.add(elapsedBatchMs);
                         messageSizeHistogram.add(messageBytes);
                         bytesConsumed += messageBytes;
+                        if (messagesConsumed >= maxMessages)
+                            break;
+
                         throttle.increment();
                     }
                     startBatchMs = Time.SYSTEM.milliseconds();
                 }
             } catch (Exception e) {
+                // TODO: Should we close task on consumer failure?
```

Comment:
This is an open question - I'm not sure what's better - to abort the whole task if one consumer fails or continue running until the last one fails

Reply:
Makes sense. I have logic implemented that does not fail the whole task is a `StatusUpdater` of one consumer fails - only when all `StatusUpdaters` fail does the task get aborted. Thoughts on that?

Reply:
The status updaters should not fail, right?  Any failure should abort the worker.

Reply:
Seems reasonable. I reverted my commit